client/header/activity-panel/panels/orders.js

```diff
@@ -266,28 +275,60 @@ export default compose(
 		}
 
 		if ( hasActionableOrders ) {
+			// Query the core Orders endpoint for the most up-to-date statuses.
+			const actionableOrdersQuery = {
+				page: 1,
+				per_page: QUERY_DEFAULTS.pageSize,
+				status: orderStatuses,
+				_fields: [ 'id', 'date_created_gmt', 'status' ],
+			};
+			const actionableOrders = Array.from( getItems( 'orders', actionableOrdersQuery ).values() );
+			const isRequestingActionable = isGetItemsRequesting( 'orders', actionableOrdersQuery );
+
+			if ( isRequestingActionable ) {
+				return {
+					isError: Boolean( getItemsError( 'orders', actionableOrdersQuery ) ),
+					isRequesting: isRequestingActionable,
+					orderStatuses,
+				};
+			}
+
+			// Retrieve the Order stats data from our reporting table.
 			const ordersQuery = {
 				page: 1,
 				per_page: QUERY_DEFAULTS.pageSize,
-				status_is: orderStatuses,
 				extended_info: true,
+				order_includes: map( actionableOrders, 'id' ),
 			};
 
-			const orders = getReportItems( 'orders', ordersQuery ).data;
+			const reportOrders = getReportItems( 'orders', ordersQuery ).data;
 			const isError = Boolean( getReportItemsError( 'orders', ordersQuery ) );
 			const isRequesting = isReportItemsRequesting( 'orders', ordersQuery );
+			let orders = [];
+
+			if ( reportOrders && reportOrders.length ) {
+				// Merge the core endpoint data with our reporting table.
+				const actionableOrdersById = keyBy( actionableOrders, 'id' );
+				orders = reportOrders.map( order =>
+					merge( {}, order, actionableOrdersById[ order.order_id ] || {} )
+				);
+			}
 
 			return { orders, isError, isRequesting, orderStatuses };
 		}
 
+		// Get a count of all orders for messaging purposes.
+		// @todo: add a property to wcSettings for this?
```

Comment:
Minor nitpick: todos were update to format:

`@todo Thing to do.`

Reply:
Amended the last commit to fix this. Thanks.